### DISCUSSION TITLE
Minimum SSD Size on Mainnet When Using --pprof --prune htc --prune.r.before=11184524

### DISCUSSION THREAD

**Discussion by codesport** [erigontech/erigon#3293] [Q&A] [answered]
Is it possible to bring up an erigon node using `--pprof --prune htc --prune.r.before=11184524` with a **< 1 terabyte SSD**?   

To be specific, the SSD would only have **930 GB of real and usable free space** dedicated for erigon's datadir.  

**Use case:** live staking on Ethereum's mainnet.

**Comment by yorickdowne** [comment, marked as answer]
Yes that’s possible.

In my testing Erigon took 635 GiB to start and grew by 1 GiB a week. Not very recent but afaik DB handling shouldn’t have changed drastically. https://eth-docker.net/docs/Usage/ResourceUsage

Now that gets tight for sure with maybe 930 GiB usable to start on a 1TB SSD.

My back of the napkin said it’d last for up to 2 years, and then it’s time for that 2TB SSD - unless EIP-4444 landed by then, which sounds likely, and the 1TB keeps being enough.

And of course there’s yet another way to stretch that SSD: ZFS with zstd or lz4, which results in a 1.7 to 2x compression of Erigon’s DB. ZFS out of the box with recordsize 128kb will be slow: In my testing it’s completely fine as long as the dataset recordsize is set to 16kb, not the default 128kb.

**Reply by codesport** [reply]
Thanks for the detailed answer. 

Out of curiosity, any idea on the logic of using block 11184524 as opposed to say, 11184528 `--prune.r.before=11184524?`  

Perhaps a possible Network Upgrade event occurred at block 11184524?

**Reply by yorickdowne** [reply]
Consensus clients start checking for deposits to the deposit contract from 11184524. For eth_getLogs to return valid results, that’s the cutoff point therefore.

This will change if / when EIP-4444 is adopted.
